### PR TITLE
feat(DTFS2-6505): Fix vulnerable forms/endpoints on GEF UI

### PR DESCRIPTION
**Pull request by ChristianMcc**
## Introduction
Sanitise user input across form submissions in GEF UI that makes requests to the portal-api micro-service.

## Resolution
Added the constructPayload function from portal to GEF including tests. Used this as well as adding to already created custom payloads when sending data to the portal API from the body of requests from submitting forms in GEF

### CODE REVIEW

File: gef-ui/server/utils/constructPayload.test.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    const returned = constructPayload(mockBody, payloadProperties);
+
+    expect(expected).toEqual(returned);
+  });
+
+  it('Should return expected payload as per `properties` and without CSRF argument with data in standard body', () => {
+    const expected = {
+      currentPassword: 'AbC!2345',
+      password: 'AbC!23456',
+      passwordConfirm: 'AbC!23456',
```
Comment (abhi-markan) [inline gef-ui/server/utils/constructPayload.test.js:88]: Can we add an assertion with following:

```javascript
 const expected = {
      currentPassword: 'AbC!2345',
      password: 'AbC!23456',
      passwordConfirm: 'AbC!23456',
      roles: {
      'admin',
      'maker',
      'checker'
      }
};
```

File: portal/server/helpers/constructPayload.js [hunk 2]
```diff
@@ -25,13 +25,11 @@ const constructPayload = (body, properties, csrf = true) => {
   // Property insertion
   properties
     .filter((property) => property in body)
-    .map((property) => {
+    .forEach((property) => {
```
Comment (abhi-markan) [inline portal/server/helpers/constructPayload.js:28]: Is `forEach` expensive on execution isn't?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, few comments to address please.

File: portal/server/helpers/constructPayload.js (hunk 2 above)
Comment (IgnatG) [reply portal/server/helpers/constructPayload.js:28]: happy for us to keep `.map`

Comment (ChristianMcc) [reply portal/server/helpers/constructPayload.js:28]: From what I've read foreach is a bit faster than map in this kind of use case but happy to revert

File: gef-ui/server/utils/constructPayload.test.js (hunk 1 above)
Comment (ChristianMcc) [reply gef-ui/server/utils/constructPayload.test.js:88]: Sorry not quite understanding this one - do yo want an extra `payloadPropertiesWithRoles` array and to test that that now pulls through the roles from the mockExtraBody?

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**